lib/ansible/module_utils/vmware.py

```diff
@@ -213,6 +213,16 @@ def find_host_portgroup_by_name(host, portgroup_name):
     return None
 
 
+def _get_vm_prop(vm, expression):
+    """Safely get a property or return None"""
+    result = None
+    try:
+        result = eval(expression)
+    except:
+        pass
+    return result
```

Comment:
pass doesn't advance result to the next layer of the object, though.  Imagine a data structure like this:
``` python
class Alice:
    foo = 1

class Bob:
     bar = Alice()
     foo = 2

obj = Bob()
del obj.bar
result = _get_vm_props(obj, ('bar', 'foo')
# with return None:
result == None
# with pass
result == 2
```

For snapshot, what are you calling?  This seems right and it seems like it would do the right thing:
```
snapshot = _get_vm_prop(vm, ['snapshot'])
```